daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -662,14 +666,20 @@ object Main {
   val blobSuffix = ".bin"
 
   def getInfosetOutputter(infosetType: InfosetType.Type, os: java.io.OutputStream)
-  : Either[InfosetOutputter, DaffodilParseOutputStreamContentHandler] = {
+  : Either[InfosetOutputter, ContentHandler] = {
     val outputter = infosetType match {
       case InfosetType.XML => Left(new XMLTextInfosetOutputter(os, pretty = true))
       case InfosetType.SCALA_XML => Left(new ScalaXMLInfosetOutputter())
       case InfosetType.JSON => Left(new JsonInfosetOutputter(os, pretty = true))
       case InfosetType.JDOM => Left(new JDOMInfosetOutputter())
       case InfosetType.W3CDOM => Left(new W3CDOMInfosetOutputter())
-      case InfosetType.SAX => Right(new DaffodilParseOutputStreamContentHandler(os, pretty = true))
+      case InfosetType.EXI => {
+        val exiFactory = DefaultEXIFactory.newInstance()
+        exiFactory.getFidelityOptions.setFidelity(FidelityOptions.FEATURE_PREFIX, true)
+        val exiResult = new EXIResult()
```

Comment:
Is this also where we would turn on other options for the EXI encoder, like telling it to compress the data to minimize its size?

Reply:
Yes, the options I believe go onto the EXIFactory.